### ISSUE TITLE
.buffer drops last buffer on source complete

### ISSUE THREAD

**Issue by JaminFarr** [ReactiveX/rxjs#2154]
**RxJS version:** 5.0.0-rc.4

**Code to reproduce:**
```javascript
const tick = Observable.interval(100).take(15)

tick.buffer(Observable.interval(1000))
  .subscribe((x) => console.log(x))
```

**Expected behavior:**
```javascript
[0, 1, 2, 3, 4, 5, 6, 7, 8]
[9, 10, 11, 12, 13, 14]
```

**Actual behavior:**
```javascript
[0, 1, 2, 3, 4, 5, 6, 7, 8]
```

**Additional information:**
Operators .bufferTime, .bufferCount, .bufferWhen and .bufferToggle all immediately emit any open buffers when the source completes. Operator .buffer does not. The lose of values is unexpected.

I would expect the results of the following to be the same
```javscript
source$.buffer(closer$)
  .subscribe((x) => console.log(x))


source$.window(closer$)
  .mergeMap(x => x.toArray())
  .subscribe((x) => console.log(x)) // Outputs expected behavior
```

**Comment by JaminFarr** [comment]
If this is confirmed as a bug I'd like to have a crack at fixing it

**Comment by kwonoj** [comment]
closed by #2174 .
